Snakefile

```diff
@@ -153,21 +155,113 @@ rule iq_tree:
         rooted="output/strains/{strain}/besttree.nwk",
         unrooted=temp("output/strains/{strain}/besttree.unrooted.treefile"),
         iqtree=temp("output/strains/{strain}/besttree.unrooted.iqtree"),
-        ckp=temp("output/strains/{strain}/besttree.unrooted.ckp.gz")
+        ckp=temp("output/strains/{strain}/besttree.unrooted.ckp.gz"),
     log:
         "output/strains/{strain}/besttree.unrooted.log"
     params:
         enabled=config['iqtree']['enabled'],
         mode=config['iqtree']['mode'],
         model=config['iqtree']['model'],
-        prefix="output/strains/{strain}/besttree.unrooted"
+        prefix="output/strains/{strain}/besttree.unrooted",
+        temp="output/strains/{strain}/temp_align_variants.aln"
+
     conda:
         config["poppipe_location"] + "/envs/iqtree.yml"
     threads:
-        16
+        4
     script:
         config["poppipe_location"] + "/scripts/run_iqtree.py"
 
+rule gubbins:
+    input:
+        alignment="output/strains/{strain}/align_variants.aln",
+        start_tree="output/strains/{strain}/besttree.nwk"
+    output:
+        final_tree="output/strains/{strain}/gubbins.final_tree.tre",
+    group:
+        "gubbins"
+    # log:
+    #     "logs/gubbins_{strain}.log"
+    params:
+        prefix=config['gubbins']['prefix'],
+        tree_builder=config['gubbins']['tree-builder'],
+        min_snp=config['gubbins']['min-snps'],
+        min_window=config['gubbins']['min-window-size'],
+        max_window=config['gubbins']['max-window-size'],
+        iterations=config['gubbins']['iterations'],
+        directory=config['gubbins']['dir'],
+        aln="align_variants.aln",
+        btree="besttree.nwk"
+
+    conda:
+        config["poppipe_location"] + "/envs/gubbins.yml"
+    threads:
+        4
+    shell:
+        "pushd {params.directory} &&\
+        run_gubbins.py {params.aln} --prefix {params.prefix} --starting-tree {params.btree} --threads {threads} \
+        && popd"
+
+rule bactdating:
+    input:
+        final_tree="output/strains/{strain}/gubbins.final_tree.tre",
+        metadata=config["transmission_metadata"]
+    output:
+        rds="output/strains/{strain}/bactdate_data.rds",
+        sorted="output/strains/{strain}/sorted.rds"
+    group:
+        "outbreak"
+    log:
+        "logs/bactdating_{strain}.log"
+    conda:
+        config["poppipe_location"] + "/envs/transphylo.yml"
+    threads:
+        4
+    shell:
+        "Rscript --vanilla scripts/run_bactDating.R output/strains/{wildcards.strain}/gubbins {input.metadata}  {output.sorted} {output.rds} > {log}"
```

Comment:
Might want to add `--no-save --no-restore`, I can't remember but worth checking

Reply:
I think `--vanilla` should be taking care of that already.
"The `--vanilla` argument makes sure that user-specific R settings are ignored, and that there is no saving or restoring of workspaces. This makes the script more portable to other systems." ([r-bloggers.com](https://www.r-bloggers.com/2019/11/r-scripts-as-command-line-tools/))